api/osb/store_instances_plugin.go

```diff
@@ -382,6 +382,13 @@ func (ssi *StoreServiceInstancePlugin) PollInstance(request *web.Request, next w
 		}
 
 		operationFromDB := op.(*types.Operation)
+		if response.StatusCode == http.StatusGone {
```

Comment:
i guess there is no easy way to avoid comparing status codes twice ? (once here and once on 354) ? One way I can think of would be to have a status code switch ontop and in each case process only the relevant stuff (for example 410 gone would process only DELETE op) but this would require extracting fetching op in a common func and probably a few other refactorings and I am not sure if the end result will be better

Reply:
Here's what I was thinking about: https://github.com/Peripli/service-manager/compare/fix-orphan-instances...fix-orphan-instances-v2

I think it gets kinda nasty with the triple switch and I'd rather have it like it is.

Reply:
With the proposal you still check status code twice - you have to move the top level switch  to 354 -but yes if you prefer just keep it as is